src/utils.jl

```diff
@@ -12,6 +12,7 @@ arbitrary axes.
 See also: [`imfilter`](@ref).
 """
 centered(A::AbstractArray) = OffsetArray(A, map(n->-((n+1)>>1), size(A)))
```

Comment:
It's better to solve this in terms of *values* than in terms of *types*. For example, in my lab's https://github.com/HolyLab/CenterIndexedArrays.jl it defines an array type that is centered on 0, yet these are not `OffsetArray`s. Likewise, if you have an OffsetArray wrapping some other array type whose indexes don't start with 1, your fix will give the wrong answer.

So rather than defining the new method below, the best option would be to redefine this method so that it doesn't implicitly assume that `A`'s indexes start with 1.

Reply:
Is there any efficient way to get the latent Array for general arrays (e.g., ReshapedArray, SubArray, OffsetArray) compared to reconstructing a new one: `eltype(A).(A)` ?

`parent(::OffsetArray)` works for this purpose but `parent(::SubArray)` and `parent(reshape(1:9, 3, 3))` doesn't.

Reply:
I'm having a tendency to dispatches everything, without being aware of such an explicit solution 😂 Thanks for pointing it out.